BMS_Receiver/BMS_Receiver.cpp

```diff
@@ -0,0 +1,146 @@
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/foreach.hpp>
+
+#include "BMS_Receiver.h"
+#include <algorithm>
+
+vector<BmsParamters>& CBMSReceiver::GetBmsDataContainer()
+{
+    return m_bmsDataContainer;
+}
+BmsParamStats& CBMSReceiver::GetParameterStats()
+{
+    return m_paramstat;
+}
+
+bool CBMSReceiver::ParseDataFromConsole(string data, BmsParamters& params)
+{
+    std::stringstream ss;
+    ss.str(data);
+
+    try
+    {
+        boost::property_tree::ptree pt;
+	    boost::property_tree::read_json(ss, pt);
+    	params.jsonData = data;
+    	params.temperature = pt.get<float>("temperature");
+    	params.stateOfCharge = pt.get<float>("soc");
+    	params.chargeRate = pt.get<float>("chargingrate");
+    }
+    catch(std::exception const& e)
+    {
+        std::cerr << e.what() << std::endl;
+        return false;
+    }
+
+    return true;
+}
+
+void CBMSReceiver::PerformMinMaxCalculation()
+{
+
+    auto data = std::minmax_element(begin(m_bmsDataContainer), end(m_bmsDataContainer), 
+                                    [](BmsParamters& p1, BmsParamters& p2)
+                                    {
+                                        return p1.temperature < p2.temperature; 
+                                    });
+
+    m_paramstat.minTemperature = data.first->temperature;
+    m_paramstat.maxTemperature = data.second->temperature;
+
+    data = std::minmax_element(begin(m_bmsDataContainer), end(m_bmsDataContainer), 
+                                    [](BmsParamters& p1, BmsParamters& p2)
+                                    {
+                                        return p1.stateOfCharge < p2.stateOfCharge; 
+                                    });
+
+    m_paramstat.minSoc = data.first->stateOfCharge;
+    m_paramstat.maxSoc = data.second->stateOfCharge;
+
+    data = std::minmax_element(begin(m_bmsDataContainer), end(m_bmsDataContainer), 
+                                    [](BmsParamters& p1, BmsParamters& p2)
+                                    {
+                                        return p1.chargeRate < p2.chargeRate; 
+                                    });
+
+    m_paramstat.minChargeRate = data.first->chargeRate;
+    m_paramstat.maxChargeRate = data.second->chargeRate;
+}
+
+static float MovingAvg(float* ptrArrNum, float* ptrSum, int pos, int len, float nextNum)
+{
+    //Subtract the oldest number from the prev sum, add the new number
+    *ptrSum = *ptrSum - ptrArrNum[pos] + nextNum;
+    //Assign the nextNum to the position in the array
+    ptrArrNum[pos] = nextNum;
+    //return the average
+    return *ptrSum / len;
+}
+
+void CBMSReceiver::PerformSimpleMovingAverage()
+{
+    int pos = 0;
+    float sum, tTemp, tSoc, tCr;
```

Comment:
`sum` is not required? You can pass the address of `tTemp, tSoc, tCr` directly?
Try naming them better, so that your intent is clear. `tTemp` doesn't convey your intention here.